internal/state/api/members.go

```diff
@@ -176,3 +177,85 @@ func (s *Server) getUser(w http.ResponseWriter, r *http.Request, p httprouter.Pa
 
 	return s.writeTerm(w, m)
 }
+
+func (s *Server) getUsers(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+	// id1,id2,id3
+	userIDs := r.URL.Query().Get("userIDs")
+	
+	if userIDs == "" {
+		return xerrors.Errorf("empty userids array")
+	}
+
+	userIDsSlice := strings.Split(userIDs, ",")
+
+	var convertedUserIDs []int64
+
+	for _, userId := range userIDsSlice {
+		value, err := strconv.ParseInt(userId,10,64);
+		if err != nil {
+			return xerrors.Errorf("parse int for userids array: %w", err)
+		}
+		convertedUserIDs = append(convertedUserIDs,value)
+	}
+
+	data, err := s.db.GetUsersDiscordIdAndUsername(r.Context(), convertedUserIDs)
+	if err != nil {
+		return xerrors.Errorf("read users: %w", err)
+	}
+
+	result := make([][]byte, len(data))
+
+    for i, userAndData := range data {
+        // Convert each UserAndData to []byte
+        userAndDataBytes, err := json.Marshal(userAndData)
+        if err != nil {
+            return xerrors.Errorf("converting user ids and discord data to [][]byte: %w", err)
+        }
+
+        result[i] = userAndDataBytes
+    }
+
+	return s.writeTerms(w, result)
+}
+
+// Define a handler function for SetGuildMembers
+func (s *Server) setGuildMembers(w http.ResponseWriter, r *http.Request, p httprouter.Params) error {
+    // Parse guild ID from the URL parameter
+	guild, err := guildParam(p)
+	if err != nil {
+		return xerrors.Errorf("read guild param: %w", err)
+	}
+	
+	var guildMemberData = make(map[string][]byte)
+    if err := json.NewDecoder(r.Body).Decode(&guildMemberData); err != nil {
+        http.Error(w, "Invalid JSON", http.StatusBadRequest)
+        return xerrors.Errorf("parse body json: %w", err)
+    }
+
+	var convertedGuildMemberData = make(map[int64][]byte)
+
+	for key,value := range guildMemberData {
+		num, err := strconv.ParseInt(key,10,64);
+		if err != nil {
+			return xerrors.Errorf("convert discord id from string to int64: %w", err)
+		}
+
+		memberJSON, err := json.Marshal(value);
```

Comment:
the `value` already []byte, no need to json.Marshal here.

Reply:
Oh ok I thought Marshal was not just convert, but also format.

Reply:
I was wrong about this, I changed it from []byte to interface{} again instead of GuildMember struct. It is probably due to the format of data I am passing in. I think using the interface{} is acceptable